File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-15.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> counts;
        int l = 0, res = 0, most = 0;
        for (int r = 0; r < s.length(); ++r) {
            counts[s[r]]++;
            most = max(most, counts[s[r]]);
            while (r - l + 1 - most > k) {
                counts[s[l]]--;
                l++;
            }
            res = max(res, r - l + 1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-15.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res, l = 0, 0
        for r, p in enumerate(prices):
            if p < prices[l]:
                l = r
            res = max(res, prices[r] - prices[l])
        return res
